lib/Runtime/Debug/DebugContext.cpp

```
@@ -142,11 +142,7 @@ namespace Js
         });
         END_TRANSLATE_OOM_TO_HRESULT(hr);
 
-        if (hr != S_OK)
-        {
-            Assert(FALSE);
-            return hr;
-        }
+        DEBUGGER_ATTACHDETACH_FATAL_ERROR_IF_FAILED(hr);
```

Yes we should return OOM and let the caller which have fail fast handle it